src/lib/combinators.rs

```diff
@@ -454,9 +454,54 @@ pub fn fold<T: Copy, I, O, S, P: Parser<I, O, S>>(acc: T,
     })
 }
 
+/// Combine two parsers into a parser that returns the result of the parser
+/// that consumed the most input.
+///
+/// If both parsers consume the same amount of input, the result of the first
+/// parser will be chosen.
+///
+/// If only one of the parsers succeeds, its result will be returned regardless
+/// of the consumed size of either parser.
+///
+/// ### Arguments
+/// * `p1` - the first parser
+/// * `p2` - the second parser
+#[inline]
+pub fn greedy_or<I: SliceLike, S, O>(p1: impl Parser<I, O, S>,
+                                     p2: impl Parser<I, O, S>
+) ->  impl Parser<I, O, S> {
+    create_parser!(s, {
+        let pos = s.input;
+        let res1 = p1(s);
+        let p1_consumed = pos.slice_len() - s.input.slice_len();
+        let p1_pos = s.input;
+
+        s.input = pos;
+        let res2 = p2(s);
+        let p2_consumed = pos.slice_len() - s.input.slice_len();
+        let p1_is_some = res1.is_some();
```

Comment:
meh, missed the `s.input = p1_pos` for the 2nd branch

Reply:
Your suggestion is not equivalent, since it doesn't select the result of the most consuming parser in the case where both parsers fail (it always "selects" p2).

I also wanted to avoid repeating myself when doing the `s.input = p1_pos`.